scripts/kind-e2e/lib_operator_deploy_subm.sh

```diff
@@ -0,0 +1,368 @@
+  kubectl config use-context $context
+
+  kubectl label node $context-worker "submariner.io/gateway=true" --overwrite
+}
+
+function create_subm_clusters_crd() {
+  # Accept cluster context as param
+  context=$1
+  kubectl config use-context $context
+
+  pushd $subm_op_dir
+
+  clusters_crd_file=deploy/crds/submariner_clusters_crd.yaml
+
+  # TODO: Can/should we create this with Op-SDK?
+cat <<EOF > $clusters_crd_file
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: clusters.submariner.io
+spec:
+  group: submariner.io
+  version: v1
+  names:
+    kind: Cluster
+    plural: clusters
+  scope: Namespaced
+EOF
+
+  cat $clusters_crd_file
+
+  # Create clusters CRD
+  # NB: This must be done before submariner-engine pod is deployed
+  if ! kubectl get crds | grep clusters.submariner.io; then
+    kubectl create -f $clusters_crd_file
+  fi
+
+  popd
+}
+
+function create_subm_endpoints_crd() {
+  # Accept cluster context as param
+  context=$1
+  kubectl config use-context $context
+
+  pushd $subm_op_dir
+
+  endpoints_crd_file=deploy/crds/submariner_endpoints_crd.yaml
+
+  # TODO: Can/should we create this with Op-SDK?
+cat <<EOF > $endpoints_crd_file
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: endpoints.submariner.io
+  annotations:
+spec:
+  group: submariner.io
+  version: v1
+  names:
+    kind: Endpoint
+    plural: endpoints
+  scope: Namespaced
+EOF
+
+  cat $endpoints_crd_file
+
+  # Create endpoints CRD
+  # NB: This must be done before submariner-engine pod is deployed
+  if ! kubectl get crds | grep endpoints.submariner.io; then
+    kubectl create -f $endpoints_crd_file
+  fi
+
+  popd
+}
+
+function create_routeagents_crd() {
+  # Accept cluster context as param
+  context=$1
+  kubectl config use-context $context
+
+  pushd $subm_op_dir
+
+  routeagents_crd_file=deploy/crds/submariner_routeagents_crd.yaml
+
+  # TODO: Can/should we create this with Op-SDK?
+cat <<EOF > $routeagents_crd_file
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: routeagents.submariner.io
+  annotations:
+spec:
+  group: submariner.io
+  version: v1alpha1
+  names:
+    kind: Routeagent
+    plural: routeagents
+  scope: Namespaced
+EOF
+
+  cat $routeagents_crd_file
+
+  # Create routeagents CRD
+  if ! kubectl get crds | grep routeagents.submariner.io; then
+    kubectl create -f $routeagents_crd_file
+  fi
+
+  popd
+}
+
+function deploy_subm_operator() {
+  # Accept cluster context to deploy SubM Operator into as param
+  context=$1
+  kubectl config use-context $context
+
+  pushd $subm_op_dir
+
+  # If SubM namespace doesn't exist (ignore SubM Broker ns), create it
+  if ! kubectl get ns | grep -v $subm_broker_ns | grep $subm_ns; then
+    # TODO: Make this dynamically use any $subm_ns
+    kubectl create -f deploy/namespace.yaml
+  fi
+
+  if ! kubectl get crds | grep submariners.submariner.io; then
```

Comment:
Yes, but this would be an example of things actually becoming more complex from creating "master" DRY functions - can't be much more clear/easy to understand than this simple one-liner on a file created by the SDK.